pkg/apis/performanceprofile/v2/performanceprofile_validation_test.go

```diff
@@ -369,6 +369,62 @@ var _ = Describe("PerformanceProfile", func() {
 			})
 		})
 	})
+
+	Describe("validation of validateFields function", func() {
+		It("should check all fields", func() {
+			// config all specs to rise an error in every func inside validateFields()
+			reservedCPUs := CPUSet("")
+			isolatedCPUs := CPUSet("0-6")
+			offlinedCPUs := CPUSet("7")
+			profile.Spec.CPU.Reserved = &reservedCPUs
+			profile.Spec.CPU.Isolated = &isolatedCPUs
+			profile.Spec.CPU.Offlined = &offlinedCPUs
+
+			profile.Spec.MachineConfigLabel["foo"] = "bar"
+
+			incorrectDefaultSize := HugePageSize("!#@")
+			profile.Spec.HugePages.DefaultHugePagesSize = &incorrectDefaultSize
+
+			profile.Spec.WorkloadHints = &WorkloadHints{
+				RealTime: pointer.BoolPtr(false),
+			}
+			profile.Spec.RealTimeKernel = &RealTimeKernel{
+				Enabled: pointer.BoolPtr(true),
+			}
+
+			invalidVendor := "123"
+			invalidDevice := "0x12345"
+			profile.Spec.Net.Devices[0].InterfaceName = pointer.StringPtr("")
+			profile.Spec.Net.Devices[0].VendorID = pointer.StringPtr(invalidVendor)
+			profile.Spec.Net.Devices[0].DeviceID = pointer.StringPtr(invalidDevice)
+
+			errors := profile.validateFields()
+
+			errorMsgs := map[string]bool{}
```

Comment:
this works. The golang idiom to implement a set is to use `struct{}` as element because it enables better compiler optimizations, possibly avoiding to use any space for the value element.
Alternatively, use `https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets`

Reply:
Done with set